src/Language/PureScript/AST/Exported.hs

```diff
@@ -144,3 +141,16 @@ isDctorExported ident (Just exps) ctor = test `any` exps
   test (TypeRef _ ident' Nothing) = ident == ident'
   test (TypeRef _ ident' (Just ctors)) = ident == ident' && ctor `elem` ctors
   test _ = False
+
```

Comment:
> I suppose we could make a note of the original export lists before desugaring so that we don't have to make any changes to Sugar.Names

I think this is the way to go. This would also allow doing things like sectioning by putting doc comments in between the exports like haddock allows. `psci` could just run the same ordering code as docs, which would also allow the sections to appear within the repl (I think that would be nice).